tls/s2n_handshake_transcript.c

```diff
@@ -143,6 +82,12 @@ int s2n_conn_update_handshake_hashes(struct s2n_connection *conn, struct s2n_blo
         GUARD(s2n_hash_update(&conn->handshake.sha512, data->data, data->size));
     }
 
+    /* Copy the CLIENT_HELLO -> SERVER_FINISHED hash.
+     * We'll need it later to calculate the application secrets. */
+    if (s2n_conn_get_current_message_type(conn) == SERVER_FINISHED) {
+        GUARD(s2n_tls13_conn_copy_server_finished_hash(conn));
```

Comment:
Does this need to be called in other TLS versions? Should we check `conn->actual_protocol_version >= S2N_TLS13`?

Reply:
Good catch. There's no reason to copy the hash pre-TLS1.3.